packages/typespec-vscode/src/extension.ts

```diff
@@ -276,3 +289,29 @@ function showStartUpMessages(stateManager: ExtensionStateManager) {
     stateManager.cleanUpStartUpMessage(workspaceFolder.uri.fsPath);
   });
 }
+
+async function shendEmitterTelemetryData(activityId: string) {
+  const uris = await TraverseTspConfigFileInWorkspace();
+  if (uris.length === 0) {
+    logger.debug(
+      "No tspconfig.yaml file found in workspace.Therefore, it is impossible to send telemetry data of emitter",
+    );
+    return;
+  }
+
+  for (const uri of uris) {
+    const doc = await vscode.workspace.openTextDocument(uri);
+    const yaml = tryParseYaml(doc.getText());
+    if (yaml) {
+      const emitTarget = yaml.contents.get("emit");
+      emitTarget.items.map((item: any) => {
```

Comment:
it's better to collect all the emitters, distinct and then send them in foreach to avoid sending unnecessary dup data